streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamsMetadataState.java

```diff
@@ -127,6 +137,7 @@ public String toString(final String indent) {
      * @return The {@link StreamsMetadata} for the storeName and key or {@link StreamsMetadata#NOT_AVAILABLE}
      * if streams is (re-)initializing
      */
+    @Deprecated
```

Comment:
Agree. Will point out alternative.. I dont think we should we mix version information here, since we could backport something, this may be distributed differently etc. 
 KafkaStreams#close method for e.g does not do that.